alsi/mappings.py

```diff
@@ -0,0 +1,102 @@
+from enum import Enum
+from typing import Union
+from .exceptions import InvalidCountryException
+
+
+def retrieve_country(input: Union["Area", str]) -> "Area":
+
+    if isinstance(input, Area):
+        return input
+    if isinstance(input, str):
+        if Area.has_member(input.upper()):
+            return Area[input.upper()]
+
+        check_country = [
+            area
+            for area in Area
+            if area._country_name.lower() == input.lower()
+        ]
+
+        if check_country:
+            return check_country[0]
+        else:
+            raise InvalidCountryException("Invalid Country")
```

Comment:
The method `retrieve_country` returns `None` if `input` is neither `Area` nor `str`, which means the return type should be `Optional[Area]`. However, instead of changing the return type, please raise an exception to signal a wrong argument.